test/test_clientagent.py

```diff
@@ -3028,5 +3038,29 @@ def test_get_network_address(self):
 
         self.server.send(Datagram.create_remove_channel(10052))
 
+    def test_heartbeat_timeout(self):
+        # Test the interest timeout
+        self.server.flush()
+        client = self.connect(port = 51201)
+        id = self.identify(client)
+
+        # Bring client out of the sandbox
+        self.set_state(client, CLIENT_STATE_ESTABLISHED)
+
+        # Send a heartbeat
+        dg = Datagram()
+        dg.add_uint16(CLIENT_HEARTBEAT)
+        client.send(dg)
+
+        # The client shouldn't hear anything back.
+        self.expectNone(client)
+
+        time.sleep(0.75)
```

Comment:
Ah, I see how it passes! The expect is waiting. Do we have any check implemented to check for open connections?